Assignment1/assignment2/lib/utilities.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:assignment2/users.dart';
+import 'dart:collection';
+
+Map<int, User> currentSession = Map();
+
+void nameHandler(String name) {
+  if (name.length == 0 || name.contains('[0-9]')) {
+    throw Exception(
+        "Name entered isn't valid, please enter appropriate details");
+  }
+}
+
+void ensurePositive(int number) {
+  if (number < 0) {
+    throw Exception(
+        "Number entered is not valid, please enter a positive number");
+  }
+}
+
+void ensureRollNumber(int rollNumber) {
+  if (currentSession.containsKey(rollNumber)) {
+    throw Exception(
+        "Roll Number already exists, please enter a new roll number");
+  }
+}
+
+List<String> acceptCourses(String? courses) {
+  List<String> registered = courses!.split(" ");
+  if (registered.length != 4) {
+    throw Exception("Registered courses should mandatorily be 4");
+  }
+  List<String> registered_courses = [];
+  registered.forEach((element) {
+    registered_courses.add(element.toUpperCase());
+  });
+  List<String> allowedCourses = ["A", "B", "C", "D", "E", "F"];
```

Comment:
This can be moved to be a constant since this won't change

Reply:
done